index-advisor/standalone/settings.mdx

```diff
@@ -0,0 +1,153 @@
+* `relpages`: Table size (counted in 8kb pages) determined by the last analyze/vacuum run
+* `curpages`: Current table size on disk (counted in 8kb pages)
+
+Index settings must be written as:
+
+```
+SET pganalyze.[settingname].[schemaname].[indexname] = [value];
+```
+
+For indexes you can set the following parameters:
+
+* `reltuples`: Index tuple count determined by the last analyze/vacuum run (typically the same as the table itself, except for partial indexes)
+* `relpages`: Index size (counted in 8kb pages) determined by the last analyze/vacuum run
+* `curpages`: Current index size on disk (counted in 8kb pages)
+* `btrootheight`: Location of the B-tree root page in the index, defaults to 1 (only applicable for B-tree index types)
+
+Attribute settings must be written as:
+
+```
+SET pganalyze.[settingname].[schemaname].[tablename].[attributename] = [value];
+```
+
+For table attributes you can set the following parameters (based on `pg_stats`):
+
+* `avg_width`: Average width in bytes of column's entries
+* `null_frac`: Fraction of column entries that are null
+* `n_distinct`: If greater than zero, the estimated number of distinct values in the column. If less than zero, the negative of the number of distinct values divided by the number of rows. 
+* `correlation`: Statistical correlation between physical row ordering and logical ordering of the column values.
+* `most_common_vals`: A list of the most common values in the column, represented as an array (`'{"value1", "value2"}'`)
+* `most_common_freqs`: A list of the frequencies of the most common values, i.e., number of occurrences of each divided by total number of rows, represented as an array (`'{0.24, 0.13}'`)
+* `histogram_bounds`: A list of values that divide the column's values into groups of approximately equal population, represented as an array (`'{"value1", "value2"}'`)
+
+In addition, all table attributes can also be suffixed by `_inherited` to set statistics for partition table parents instead (which include child partitions).
+
+## Automatically generating SET statements for table statistics
+
+You can run the following SQL statement to automatically generate the appropriate `SET` statements for all
+tables in your database.
+
+Note that using the `pgstattuple` and `pageinspect` extensions is optional. Since most common values can be sensitive, e.g.
+for a password column, the most common values data is not collected for array/composite/string or user-defined types.
+
+You should audit the resulting SET statements for any sensitive information, and especially consider numeric columns that
+could contain sensitive data. For example, if you have salary or revenue data stored in your database, the `pg_stats` data may
+contain samples of that data.
+
+```sql
+CREATE EXTENSION IF NOT EXISTS pgstattuple; --- For pg_relpages (also uncomment the relevant lines below)
+CREATE EXTENSION IF NOT EXISTS pageinspect; --- For bt_metap (also uncomment the relevant line below)
```

Comment:
Ideally we should dynamically update the query via checkboxes or something for these, but this is totally fine for now.

Reply:
Yeah, agreed - something to improve another day :)